### DISCUSSION TITLE
Timestamp in chat for response completion, not just initiation

### DISCUSSION THREAD

**Discussion by sengenbe** [open-webui/open-webui#1494] [General]
**Is your feature request related to a problem? Please describe.**
The timestamp used in chat responses seems to be when the response is initiated, not when complete.  For example, the timestamp when interrogating a document matches the prompt timestamp even though the actual response content takes several minutes to generate given hardware constraints.  This makes it difficult to benchmark.

**Describe the solution you'd like**
Have a timestamp showing the completed response time

**Describe alternatives you've considered**
Counting in my head

**Additional context**
None

**Comment by sengenbe** [comment]
Edit: To be even more direct, ideally the reponse would provide the length of time it took to generate the response

**Comment by justinh-rahb** [comment]
> Edit: To be even more direct, ideally the reponse would provide the length of time it took to generate the response

It does:

![Screenshot 2024-04-10 at 2 10 14 PM](https://github.com/open-webui/open-webui/assets/52832301/d4cf059c-fbf1-459c-a3bc-fb309fc1868f)

I realize it is extra work but can you not take the initiation timestamp and add the duration to get the completion timestamp?

**Reply by sengenbe** [reply]
Ah! I was not aware of the info box and was referring to the timestamp shown next to the model name (e.g. Mistral 7b 10/4/2024 13:55).  Yep doing that math would do the trick.  Seems a bit more logical to present the final completion time in the hover area next to the model name but understand there are likely use cases for what's there now.
